tests/manage/noobaa/conftest.py

```diff
@@ -44,31 +44,51 @@ def object_cleanup():
                 command=craft_s3_command(noobaa_obj, "rm " + uploaded_filename),
                 secrets=[noobaa_obj.access_key_id, noobaa_obj.access_key, noobaa_obj.endpoint]
             )
+
     request.addfinalizer(object_cleanup)
     return uploaded_objects_paths
 
 
 @pytest.fixture()
-def created_buckets(request, noobaa_obj):
+def bucket_factory(request, noobaa_obj):
     """
-    Deletes all buckets that were created as part of the test
+    Creates and deletes all buckets that were created as part of the test
 
     Args:
         noobaa_obj (NooBaa): A NooBaa object containing the NooBaa S3 connection credentials
+        amount (int): The amount of buckets to create
 
     Returns:
         list: An empty list of buckets
 
     """
-    created_buckets_names = []
+    created_bucket_names = []
+
+    def _bucket_factory(amount=1):
+        for i in range(amount):
+            bucket_name = create_unique_resource_name(
+                resource_description='bucket', resource_type='s3'
+            )
+            logger.info(f'Creating bucket: {bucket_name}')
+            created_bucket_names.append(
+                noobaa_obj.s3_create_bucket(bucketname=bucket_name)
+            )
+        return created_bucket_names
 
     def bucket_cleanup():
-        for bucket in created_buckets_names:
-            logger.info(f'Deleting bucket {bucket.name}')
-            bucket.object_versions.delete()
-            noobaa_obj.s3_delete_bucket(bucket)
+        all_existing_buckets = noobaa_obj.s3_list_all_bucket_names()
+        for bucket in created_bucket_names:
+            if bucket.name in all_existing_buckets:
+                logger.info(f'Deleting bucket {bucket.name}')
+                bucket.object_versions.delete()
+                noobaa_obj.s3_delete_bucket(bucket)
+                logger.info(
+                    f"Verifying whether bucket: {bucket.name} exists after deletion"
+                )
+                assert noobaa_obj.s3_verify_bucket_exists(bucket) is False
```

Comment:
You can do:
assert not noobaa_obj.s3_verify_bucket_exists(bucket), f"s3 bucket {bucket} still exists"

Reply:
Logging is performed under `s3_verify_bucket_exists()`, so I kept the error out :)
Thanks!